### PR TITLE
Add sticky session support (round two!)

### PR DESCRIPTION
**Pull request by owen**
Hi @emilevauge, let's try this again :-)

This fixes the problems I was having with the Docker integration.  I'm not super happy with the way it turned out - it could probably use a refactoring.  But I've tested this and it works locally with my containers & Marathon environments.

(and this way I won't need to have a million merge commits :-))

### CODE REVIEW

**Comment by emilevauge** [comment]
@owen thanks a lot for this contribution (linked to https://github.com/containous/oxy/pull/4) :+1: 
Few comments:

- you need to rebase ;)
- you will have to change glide.yaml with the new `containous/oxy` version

thanks !

**Comment by emilevauge** [comment]
PIng @owen ?

**Comment by owen** [comment]
@emilevauge **thank you** for the reminder, this completely fell off my radar 😦

I've rebased and glide updated - that picked up some additional lock file changes as well I believe.  I can change this patch to just include the new oxy revision if you'd like.

This passes the unit & integration tests locally.

File: docs/basics.md [hunk 1]
```diff
@@ -222,6 +222,17 @@ For example:
 - Another possible value for `extractorfunc` is `client.ip` which will categorize requests based on client source ip.
 - Lastly `extractorfunc` can take the value of `request.header.ANY_HEADER` which will categorize requests based on `ANY_HEADER` that you provide.
 
+Sticky sessions are supported with both load balancers. When sticky sessions are enabled, a cookie called `_TRAEFIK_BACKEND` is set on the initial
```
Comment (emilevauge) [inline docs/basics.md]: It seems that in the code it's `_TRAEFIK_SERVERNAME` and not `_TRAEFIK_BACKEND`

Comment (owen) [inline docs/basics.md]: indeed it is - I changed the code to `_TRAEFIK_BACKEND` as that's more consistently named with the rest of the code

**Comment by emilevauge** [comment]
@owen have you tested it with Docker ? I can get it work :'(

**Comment by owen** [comment]
oh no!  i had before - let me get it built and see what's up :)

**Comment by owen** [comment]
I looked at this and was pretty stumped. The docker provider properly sets the cookie and sets up sticky sessions. I'll be digging in more today - don't want you to think I forgot again!

**Comment by dencric** [comment]
I am curious if the stick session feature is going to be available anytime soon.

**Comment by emilevauge** [comment]
@owen ping ^^ ?

**Comment by samart** [comment]
Hi all, any word on this feature?   would really like to replace what we have now with traefik...

**Comment by owen** [comment]
Oof.  Sorry all, I've been slammed.  Hoping to take a look today/this weekend.

**Comment by Russell-IO** [comment]
@owen ping

**Comment by emilevauge** [comment]
Closing this one, as being inactive, in favor of #701 
Thanks for your work @owen :)

**Pull request closed without merging**